cmd/api/src/database/saved_queries.go

```diff
@@ -71,6 +78,24 @@ func (s *BloodhoundDB) CreateSavedQuery(ctx context.Context, userID uuid.UUID, n
 	return savedQuery, CheckError(s.db.WithContext(ctx).Create(&savedQuery))
 }
 
+func (s *BloodhoundDB) UpdateSavedQuery(ctx context.Context, savedQueryID int, name, query, description string) (model.SavedQuery, error) {
+	if savedQuery, err := s.GetSavedQuery(ctx, savedQueryID); err != nil {
+		return model.SavedQuery{}, err
+	} else {
+		if name != "" {
+			savedQuery.Name = name
+		}
+		if query != "" {
+			savedQuery.Query = query
+		}
+		if description != "" {
+			savedQuery.Description = description
+		}
```

Comment:
A user might want to remove a `description` from their saved query.
It probably doesn't matter as much for `query` and `name` but it might be feel weird to a user if they're intentionally deleting a saved query's cypher and when saving it is popping back up.

I'd think we could move the empty check logic outside of the database interaction and can just pass in `model.SavedQuery` that we expect to update.
https://gorm.io/docs/update.html